xarray/testing.py

```diff
@@ -0,0 +1,95 @@
+"""Testing functions exposed to the user API"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+from xarray.core import ops
+
+
+def decode_string_data(data):
+    if data.dtype.kind == 'S':
+        return np.core.defchararray.decode(data, 'utf-8', 'replace')
+    return data
+
+
+def data_allclose_or_equiv(arr1, arr2, rtol=1e-05, atol=1e-08):
+    if any(arr.dtype.kind == 'S' for arr in [arr1, arr2]):
```

Comment:
If this is going to be public API, maybe we should hide this string decoding behind an option, e.g., `decode_bytes`? We could even still default to `decode_bytes=True`, that would at least signal that something is going on.